doc/content/microreactors/airjacket/index.md

```diff
@@ -0,0 +1,510 @@
+
+!listing microreactors/gcmr/airjacket/airjacket.usr start= rescale the domain end=enddo include-end=True
+
+Once the mesh has the desired orientation, the axisymmetric mode can be turned on in the `par` file using the `axiSymmetry = yes` line in the `[PROBLEMTYPE]` block:
+
+!listing microreactors/gcmr/airjacket/airjacket.par start=PROBLEMTYPE end=axiSymmetry include-end=True
+
+### Boundary Conditions, Initial Conditions, Boussinesq approximation
+
+Tying sidesets to a type of boundary condition can be accomplished through the `par` file. For each field, such as `VELOCITY`, sidesets
+starting from 1 can be assigned a boundary type using the `boundaryTypeMap` key:
+
+!listing microreactors/gcmr/airjacket/airjacket.par start=VELOCITY end=boundary include-end=True
+
+A similar action can be accomplished in the `usr` file, in the `usrdat2` routine, by modifying the `cbc` array. This is recommended when
+your sideset IDs have arbitrary numbers that do not start with `1`
+
+!listing microreactors/gcmr/airjacket/airjacket.usr start=assign BC end=c      enddo include-end=True
+
+The inlet boundary has a constant $u_x$, $u_y$, $k$, and $\tau$ boundary condition, and temperature is set to 0 based on our nondimensionalization.
+The hot wall temperature is set to a constant value of 1 in nondimensional terms (420 $^\circ$C), that is implemented with a short, linear ramp
+to avoid sharp gradients that create problems with resolution of scalars. The walls have zero Dirichlet conditions for $k$ and $\tau$. The outlet
+boundary condition is do-nothing, specified in the `par` file.  The boundary conditions are specified in the `usr` file in the `userbc` subroutine:
+
+!listing microreactors/gcmr/airjacket/airjacket.usr start=subroutine userbc end=c---- include-end=False
+
+The initial conditions are specified in `useric` as follows
+
+!listing microreactors/gcmr/airjacket/airjacket.usr start=subroutine useric end=c---- include-end=False
+
+Note that these initial conditions are overwritten if the simulation is restarted from another run's output, using the `startFrom` option
+in the `par` file. A good initial condition is vital for stability and good convergence. Some tips for generating the initial condition for this
+simulation include running with a coarse mesh, turning off the axisymmetric mode, and using the obtained data to restart the simulation using the
+output obtained.
+
+The Boussinesq approximation is applied as a forcing term in the `userf` routine. Note that the Richardson number is calculated in the
+`userchk` subroutine, and transferred to `userf` using a `common` block, as shown below. Complicated calculations that require global
+operations across all MPI ranks are best performed in `userchk`, as subroutines like `userbc`, `useric`, and `userf` are not called
+by every MPI process, and attempting an operation like a global sum can cause the program to freeze. The Boussinesq term is gradually
```

Comment:
Yes, and so does Elia, and the Nek documentation. `userchk` is considered the best place for such calculations, and the documentation and `usr` file templates warn the user that subroutines like`userbc` etc are not called by every process _by design_ , so it's not meant for calculations like that. This portion of the tutorial was written to prevent such user errors, and explain to the user why this behaviour occurs.

If this reads too much like a description of a bug rather than a feature, please let me know.

Reply:
I think it could be reformulated or explained more, as you want